packages/shorebird_cli/lib/src/logging/detail_progress.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:mason_logger/mason_logger.dart';
+
+/// {@template detail_progress}
+/// A [Progress] wrapper that allows for maintaining a base message (e.g., a
+/// task title) while updating the message with more specific information,
+/// rendered in dark gray.
+/// {@endtemplate}
+class DetailProgress implements Progress {
```

Comment:
I probably wouldn't have implemented Progress, just wrapped one w/o implementing the interface.

Reply:
I think you don't use most of the impls, so no point in adding them now?

Reply:
Discussed offline